repos/fdbt-site/tests/pages/returnService.test.tsx

```diff
@@ -0,0 +1,212 @@
+        description: 'this bus service is 123',
+        origin: 'Manchester',
+        destination: 'Leeds',
+        serviceCode: 'NW_05_BLAC_123_1',
+    },
+    {
+        id: 12,
+        lineName: 'X1',
+        lineId: '3h3vb32ik',
+        startDate: '06/02/2020',
+        description: 'this bus service is X1',
+        origin: 'Edinburgh',
+        serviceCode: 'NW_05_BLAC_X1_1',
+    },
+    {
+        id: 13,
+        lineName: 'Infinity Line',
+        lineId: '3h3vb32ik',
+        startDate: '07/02/2020',
+        description: 'this bus service is Infinity Line',
+        destination: 'London',
+        serviceCode: 'WY_13_IWBT_07_1',
+    },
+];
+
+describe('pages', () => {
+    describe('returnService', () => {
+        beforeEach(() => {
+            (getServicesByNocCodeAndDataSource as jest.Mock).mockImplementation(() => mockServices);
+        });
+
+        it('should render correctly', () => {
+            const tree = shallow(
+                <ReturnService
+                    operator="Connexions Buses"
+                    passengerType="Adult"
+                    services={mockServices}
+                    error={[]}
+                    csrfToken=""
+                    selectedServiceId={1}
+                    backHref=""
+                />,
+            );
+            expect(tree).toMatchSnapshot();
+        });
+
+        it('shows operator name above the select box', () => {
+            const wrapper = shallow(
+                <ReturnService
+                    operator="Connexions Buses"
+                    passengerType="Adult"
+                    services={mockServices}
+                    error={[]}
+                    csrfToken=""
+                    selectedServiceId={1}
+                    backHref=""
+                />,
+            );
+            const operatorWelcome = wrapper.find('#service-operator-passenger-type-hint').first();
+
+            expect(operatorWelcome.text()).toBe('Connexions Buses - Adult');
+        });
+
+        it('shows a list of services for the operator in the select box with bods data source', () => {
+            const wrapper = shallow(
+                <ReturnService
+                    operator="Connexions Buses"
+                    passengerType="Adult"
+                    services={mockServices}
+                    error={[]}
+                    csrfToken=""
+                    selectedServiceId={1}
+                    backHref=""
+                />,
+            );
+            const operatorServices = wrapper.find('.service-option');
+
+            expect(operatorServices).toHaveLength(3);
+            expect(operatorServices.first().text()).toBe('123 Manchester - Leeds (Start date 05/02/2020)');
+            expect(operatorServices.at(1).text()).toBe('X1 Edinburgh - N/A (Start date 06/02/2020)');
+            expect(operatorServices.at(2).text()).toBe('Infinity Line N/A - London (Start date 07/02/2020)');
+        });
+
```

Comment:
Are we missing a snapshot test of how it renders when there are errors in the props?